src/livestatus.c

```diff
@@ -0,0 +1,463 @@
+           sizeof(vl.plugin_instance));
+  vl.values = &(value_t){.gauge = value};
+
+  return plugin_dispatch_values(&vl);
+} /* int ls_collectd_dispatch_count */
+
+static int ls_collectd_dispatch_counter(counter_t value,
+                                        const char *plugin_instance) {
+  value_list_t vl = ls_collectd_init_vl();
+
+  sstrncpy(vl.type, "counter", sizeof(vl.type));
+  sstrncpy(vl.plugin_instance, (char *)plugin_instance,
+           sizeof(vl.plugin_instance));
+  vl.values = &(value_t){.counter = value};
+
+  return plugin_dispatch_values(&vl);
+} /* int ls_collectd_dispatch_counter */
+
+#define L_DISPATCH(f, v, plinst)                                               \
+  if (f(v, plinst) < 0) {                                                      \
+    ERROR("livestatus plugin: fail to dispatch value for " plinst);            \
+    rc = -1;                                                                   \
+  }
+
+static int ls_collectd_dispatch(const livestatus_status_t *status) {
+  int rc = 0;
+
+  L_DISPATCH(ls_collectd_dispatch_count, status->cached_log_messages,
+             "cached_log_messages");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->connections, "connections");
+  L_DISPATCH(ls_collectd_dispatch_count, status->connections_rate,
+             "connections_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->forks, "forks");
+  L_DISPATCH(ls_collectd_dispatch_count, status->forks_rate, "forks_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->host_checks, "host_checks");
+  L_DISPATCH(ls_collectd_dispatch_count, status->host_checks_rate,
+             "host_checks_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->livecheck_overflows,
+             "livecheck_overflows");
+  L_DISPATCH(ls_collectd_dispatch_count, status->livecheck_overflows_rate,
+             "livecheck_overflows_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->livechecks, "livechecks");
+  L_DISPATCH(ls_collectd_dispatch_count, status->livechecks_rate,
+             "livechecks_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->log_messages,
+             "log_messages");
+  L_DISPATCH(ls_collectd_dispatch_count, status->log_messages_rate,
+             "log_messages_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->neb_callbacks,
+             "neb_callbacks");
+  L_DISPATCH(ls_collectd_dispatch_count, status->neb_callbacks_rate,
+             "neb_callbacks_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->requests, "requests");
+  L_DISPATCH(ls_collectd_dispatch_count, status->requests_rate,
+             "requests_rate");
+  L_DISPATCH(ls_collectd_dispatch_counter, status->service_checks,
+             "service_checks");
+  L_DISPATCH(ls_collectd_dispatch_count, status->service_checks_rate,
+             "service_checks_rate");
+
+  return rc;
+} /* int ls_collectd_dispatch */
+
+static int ls_read(void) /* {{{ */
```

Comment:
Does that socket allows persistent connections?

Reply:
[Yes it does](https://checkmk.com/cms_livestatus.html#Keeping%20a%20connection%20alive%20(KeepAlive)) but I didn't want to dedicate a connection in the Livestatus backend for the collectd monitoring.
A more flexible solution may be to allow the end-user to configure this and decide if he wants collectd to use a persistent connection.
Would it be better for you ?

Reply:
If reconnecting works for the backend, I'd say that's fine and would prefer to keep it simple. If establishing many new connections causes trouble, we can add such an option later.